Kudu.Services/Diagnostics/CrashDumpController.cs

```diff
@@ -0,0 +1,121 @@
+                    .Select(f => GetCrashDumpInfoFromFile(f, Request.RequestUri.GetLeftPart(UriPartial.Authority)));
+            }
+        }
+
+        [HttpGet]
+        public HttpResponseMessage Get(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Get({name})"))
+            {
+                return Request.CreateResponse(HttpStatusCode.OK, GetCrashDump(name));
+            }
+        }
+
+        [HttpDelete]
+        public HttpResponseMessage Delete(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Delete({name})"))
+            {
+                var crashDump = GetCrashDump(name);
+                FileSystemHelpers.DeleteFile(crashDump.FilePath);
+                return Request.CreateResponse(HttpStatusCode.OK);
+            }
+        }
+
+        [HttpPost]
+        public async Task<HttpResponseMessage> Analyze(string name)
+        {
+            using (_tracer.Step($"CrashDumpController.Analyze({name})"))
+            {
+                var crashDump = GetCrashDump(name);
+                var exe = new Executable(ResolveCdbPath(), _environment.RootPath, _settings.GetCommandIdleTimeout());
+                using (MemoryStream outputStream = new MemoryStream(), errorStream = new MemoryStream())
+                {
+                    // ExecuteAsync could deadlock if called from an ASP.NET thread
+                    var exitCode = await Task.Run(async () => await exe.ExecuteAsync(_tracer, $"-c \"!analyze -v;q\" -z \"{crashDump.FilePath}\"", outputStream, errorStream));
+                    if (exitCode == 0)
+                    {
+                        return Request.CreateResponse(HttpStatusCode.OK, new { output = outputStream.AsString() });
+                    }
+                    else
+                    {
+                        return Request.CreateResponse(HttpStatusCode.InternalServerError, new { error = errorStream.AsString() });
+                    }
+                }
+            }
+        }
+
+        //public for unit tests
+        public CrashDumpInfo GetCrashDump(string name)
+        {
+            var path = Path.Combine(_environment.CrashDumpsPath, name);
+            if (FileSystemHelpers.FileExists(path))
+            {
+                var fileInfo = FileSystemHelpers.FileInfoFromFileName(path);
+                return GetCrashDumpInfoFromFile(fileInfo, Request.RequestUri.GetLeftPart(UriPartial.Authority));
+            }
+            throw new HttpResponseException(HttpStatusCode.NotFound);
+        }
+
+        private static CrashDumpInfo GetCrashDumpInfoFromFile(FileInfoBase fileInfo, string baseUrl)
+        {
+            baseUrl = baseUrl.TrimEnd('/');
+            return new CrashDumpInfo
+            {
+                Name = fileInfo.Name,
+                Timestamp = fileInfo.LastWriteTime,
+                Href = new Uri($"{baseUrl}/api/crashdumps/{fileInfo.Name}"),
```

Comment:
Not really.   In that case, we should fix our build.cmd to strictly require 14 (not optional).